source/how-tos/enable-intel-tsn.rst

```diff
@@ -0,0 +1,311 @@
+
+Capsule location can be specified by SBL configuration data. SBL will update the TSN component inside the IPFW container using the specified capsule image during  next boot once firmware update is triggered.
+
+Details on how this is all generated and packaged will be discussed in the later sections.
+
+
+**TSN enable**
+
+For each platform, there is a build configuration file BoardConfig.py. User can check this build configuration  file to make sure self.ENABLE_TSN_MAC_ADDRESS is set to 1 to enable TSN MAC Address binary.
+The other TSN related configuration data is PchTsnEnable. Make sure to set this to 1 under Silicon Configuration Data.
+
+Take Tiger Lake as an example.
+
+  https://github.com/slimbootloader/slimbootloader/blob/master/Platform/TigerlakeBoardPkg/BoardConfig.py::
+
+    self.ENABLE_TSN_MAC_ADDRESS = 1
+    if self.ENABLE_TSN_MAC_ADDRESS:
+        self.TMAC_SIZE = 0x00001000
+
+    # Name | Image File     | CompressAlg |         AuthType          |                 Key File                   | Region Align   | Region Size  | Svn Info
+    # ===============================================================================================================================================================
+    ('IPFW',  'SIIPFW.bin',      ''    ,     container_list_auth_type,   'KEY_ID_CONTAINER'+'_'+self._RSA_SIGN_TYPE,          0,          0      ,        0),   # Container Header
+    ('TMAC',     ''       ,     'Lz4'  ,     container_list_auth_type,   'KEY_ID_CONTAINER_COMP'+'_'+self._RSA_SIGN_TYPE,     0,   self.TMAC_SIZE,        0),   # TSN MAC Address
+
+In the build configuration file, user can set ``self.ENABLE_TSN_MAC_ADDRESS`` to 1 to enable TSN component and adjust the size if needed. The TSN container name is``IPFW`` by default.
+Inside the IPFW container, the TSN MAC component is identified by 4 unique characters. SBL would use it when loading container and update it via firmware update.
+For TSN, the unique name is ``TMAC`` that SBL looks at while loading the TSN binary.
+
+
+Different options to update TSN container
+--------------------------------------------------------------------
+
+**1. Specify in Board Config file and rebuild SBL**
+
+
+Specify the raw TSN binary Image file (that is generated by FBU tool) name in build config file BoardConfig.py. Copy the binary into Platform/<Platform>BoardPkg/Binaries folder in SBL source code before building SBL.
+User can also specify different compression and authentication method for
+this binary as desired. Rebuild SBL successfully. Now, the TSN MAC binary will be created and embedded in the IPFW container and be part of the IFWI. When flashed onto the board, SBL will process this container.
+
+In addition, if the user decide to take this option, SBL that was rebuilt need to be stitched with TSN stitching option.
+
+**TSN Stitch Option for Soft Straps**

```

Comment:
Move lines 85 to 93 to 101 (after .. _tsn_settings: and before TSN configuration settings)?

Reply:
The flow is much better if this section is tagged along in option 1 method as it doesnt pertain to option 2. Leaving it here for now.